Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2539 @@
+//
+// Copyright 2018 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import QuartzCore
+import ArcGIS
+
+// MARK: - Time Slider Control
+
+/*
+ Controls how labels on the slider are displayed.
+ */
+public enum TimeSliderLabelMode {
+    case none
+    case thumbs
+    case ticks
+}
+
+/*
+ Playback direction when time is being animated.
+ */
+public enum PlaybackDirection {
+    case forward
+    case backward
+}
+
+/*
+ The looping behavior of the slider when time is being animated.
+ */
+public enum LoopMode {
```

Comment:
I would consider parameterizing LoopMode enum cases `repeat` and `reverse` with a PlaybackDirection. This way, you won't need to maintain a reference to both a LoopMode and PlaybackDirection variable. Example:

```swift
public enum LoopMode {
    case none
    case `repeat`(PlaybackDirection)
    case reverse(PlaybackDirection)
    
    var direction: PlaybackDirection? {
        switch self {
        case .none:
            return nil
        case .`repeat`(let dir), .reverse(let dir):
            return dir
        }
    }
}

public enum PlaybackDirection {
    case forward
    case backward
}

let mode = LoopMode.reverse(.forward)
```

Reply:
This would also rid the need for a nested `switch` statement in line 1116.

Reply:
done